core/vm/interpreter.go

```diff
@@ -112,15 +111,24 @@ func (in *Interpreter) Run(contract *Contract, input []byte) (ret []byte, err er
 	// as every returning call will return new data anyway.
 	in.returnData = nil
 
+	if in.intPool == nil {
+		in.intPool = poolOfIntPools.get()
+		defer func() {
+			poolOfIntPools.put(in.intPool)
+			in.intPool = nil
+		}()
+	}
```

Comment:
Please move this code up to the top. It should be an initialization code, currently it mixes in with the actual execution of the interpreter already.

Reply:
couldn't do that since it now depends on the declaration of `stack`. See my upcoming change.

Reply:
As we agreed, pls move this up.